crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -224,7 +244,21 @@ pub async fn checkpoint_process<A>(
 
         // (3) Create a new proposal locally. This will also allow other validators
         // to query the proposal.
-        let proposal = state_checkpoints.lock().new_proposal(committee.epoch);
+        // Only move to propose when we have the full checkpoint certificate
+        let sequence_number = state_checkpoints.lock().next_checkpoint();
+        if sequence_number > 0 {
+            // Check that we have the full certificate for the previous checkpoint.
+            // If not, we are not ready yet to make a proposal.
+            if !matches!(
+                state_checkpoints.lock().get_checkpoint(sequence_number - 1),
+                Ok(Some(AuthenticatedCheckpoint::Certified(..)))
+            ) {
+                tokio::time::sleep(timing.delay_on_local_failure).await;
+                continue;
```

Comment:
I would be curious as to whether more direct action is needed in this case in order to ensure liveness.

Reply:
This usually happens when there is a local signed checkpoint, but haven't got a quorum to form a cert yet. So once it loops back and try again, when enough validators have signed a checkpoint it will make progress